spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestSparkSessionCatalog.java

```diff
@@ -98,4 +103,20 @@ public void testLoadFunction() {
 
     // TODO: fix loading Iceberg built-in functions in SessionCatalog
   }
+
+  @Test
+  public void testDropHiveTable() throws NoSuchTableException {
```

Comment:
The test is valid and highlights the problem. However, I think the right fix would be in `HiveCatalog`. Let me check that part.

Reply:
Yeah, I think the right fix would be to ensure we only drop actual Iceberg tables in `HiveCatalog`.

Reply:
If we fix it in `HiveCatalog` , do we need to add a new option control whether to allow deletion of hive table, like the `list-all-tables` option in `listTables`?

Reply:
Also, should we fix this in `GlueCatalog` as well?

Reply:
Off topic: `GlueCatalog.listTables` does not respect the `list-all-tables` option, which may need improvement.